### PR TITLE
Do not store zero weight in `Foundation::cdf`

### PR DESCRIPTION
**Pull request by pjessesco**
#1993 
> Should this be done by users of `foundation::CDF`?

I thought it may force to make pair `if clause(if weight > 0)` with `cdf::insert()` in future usage. It doesn't seem good to me, so I chose to change `cdf::insert()` rather than change _users_ of `cdf`. 

Feedback appreciated :)

### CODE REVIEW

File: src/appleseed/foundation/math/cdf.h [hunk 1]
```diff
@@ -181,8 +181,10 @@ template <typename Item, typename Weight>
 inline void CDF<Item, Weight>::insert(const Item& item, const Weight weight)
 {
     assert(weight >= Weight(0.0));
-    m_items.push_back(std::make_pair(item, weight));
-    m_weight_sum += weight;
+    if(weight > Weight(0.0)){
```
Comment (est77) [inline src/appleseed/foundation/math/cdf.h]: Add space after if and move { to the next line.

```
if (condition)
{
    ....
}
```

**Comment by est77** [comment]
Thanks for the PR. I think it is correct. Only some very small formatting issues and we can merge.

File: src/appleseed/foundation/math/cdf.h (hunk 1 above)
Comment (pjessesco) [reply src/appleseed/foundation/math/cdf.h]: Oops, thanks for review.

**Comment by dictoon** [comment]
Looks good. Do unit tests pass?

**Comment by pjessesco** [comment]
Yes it does.

**Pull request merged**